apps/licences/src/lib/bff/business.ts

```diff
@@ -1,22 +1,49 @@
-import { CoreLogger } from "@io-ipatente/core";
+import { Patenti } from "@/generated/bff-openapi";
+import { ManagedInternalError } from "@io-ipatente/core";
+import { ZodiosError } from "@zodios/core";
+import { AxiosError } from "axios";
 
 import { getExternalApiClient } from "./client";
 
 export const retrieveLicences =
-  (logger: CoreLogger) =>
+  () =>
   async (additionalDataJWS: string, token: string, fiscalCode: string) => {
     try {
-      return await getExternalApiClient().getPuntiPatente({
+      // TEST: Simula errore DG_MOT
+      if (process.env.FORCE_DG_MOT_ERROR === "true") {
+        throw new AxiosError("DG_MOT server error (test mode)");
+      }
+
+      const response = await getExternalApiClient().getPuntiPatente({
         headers: {
           "Agid-JWT-TrackingEvidence": additionalDataJWS,
           Authorization: `Bearer ${token}`,
           codiceFiscale: fiscalCode,
         },
       });
+
+      const licences = Patenti.safeParse(response);
+
+      if (!licences.success) {
+        throw new ManagedInternalError(
+          "[DG_MOT] Validation failed: Invalid licence data from DG_MOT",
+          licences.error,
+        );
+      }
+
+      return licences.data;
```

Comment:
chore: move this check to the API controller layer.

It is responsibility of the API controller layer to verify that the output matches the response body defined in the OpenAPI. This is a special case where the response from the "third-party" API is identical to that of the "BFF" API. If the response is not valid, return an internal error